httpx/models.py

```diff
@@ -205,6 +205,13 @@ def __eq__(self, other: typing.Any) -> bool:
         return isinstance(other, (URL, str)) and str(self) == str(other)
 
     def __str__(self) -> str:
+        if self._uri_reference.userinfo:
+            userinfo = f"{self.username}:[secure]"
+            return self._uri_reference.copy_with(
+                authority=rfc3986.normalizers.normalize_authority(
+                    (userinfo, self._uri_reference.host, self._uri_reference.port)
+                )
+            ).unsplit()
         return self._uri_reference.unsplit()
```

Comment:
The `__str__` method shouldn't obscure the URL password. We should only do that for `__repr__`.

(We still want to allow explicit `str()` casting to function normally, we just don't want credentials leaking out of variables representations in tracebacks etc.)

Reply:
Thanks @tomchristie